core/trino-main/src/main/java/io/trino/execution/DropColumnTask.java

```diff
@@ -87,17 +91,53 @@ public ListenableFuture<Void> execute(
             return immediateVoidFuture();
         }
 
-        if (metadata.getColumnMetadata(session, tableHandle, columnHandle).isHidden()) {
+        List<String> field = statement.getColumn().getOriginalParts().stream().map(Identifier::getValue).skip(1).collect(toImmutableList());
```

Comment:
What's the reasoning for using `getParts` above and `getOriginalParts` here?

nit: line break after stream, map, skip

nit: would `List#sublist` help?

Reply:
I used `getParts` for the parent column name to keep the existing behavior. 
I used `getOriginalParts` for child fields because field names are case sensitive.

Reply:
Let's have a code comment

Reply:
> nit: would `List#sublist` help?

thanks for trying out, indeed it isn't more readable with sublist